### PR TITLE
Update skimNtuple2018.cpp

### PR DESCRIPTION
**Pull request by jonamotta**
Updated skimNtuple2018.cpp following the modifications already present in skimNtuple2017.cpp, in order to match the latest requirements for the skimming procedure.

### CODE REVIEW

File: test/skimNtuple2018.cpp [hunk 1]
```diff
@@ -90,8 +99,8 @@ const float DYscale_MTT_vHighPt[3] = {0.820, 1.999  , 0.897  };
 ** = then sigma_i = f_i * sigmal_LO
 ** - stitchWeight (0jet) = f_0 / (f_0 * N_inclusive)
 ** - stitchWeight (njet) = f_n / (f_n * N_inclusive + N_njets)
-*/ 
-// computed for Legacy2016, 23 Jan 2020
+*/
+// Legacy2016 (23 Jan 2020) - These are not used in the skimming since the DY sample is already split in jet bins and the weight is included in the XS
```
Comment (camendola) [inline test/skimNtuple2018.cpp]: @jonamotta they are used in the skimming both for 2016 and 2018, but not for 2017. Can you remove the comment? "These are non used..." (however they need to be updated for each production, so we'll update these numbers before the next round of skims)

**Comment by camendola** [comment]
Thanks @jonamotta ! Can you remove the file  test/._skimNtuple2018.cpp from this PR? you only need to remove it locally, then 
* `git add test/._skimNtuple2018.cpp`
* make a commit
* push it to **your** VBF_legacy_syst branch

As long as this PR is open, anything you push in your VBF_legacy_syst will also appear here.

File: test/skimNtuple2018.cpp [hunk 2]
```diff
@@ -1949,251 +2221,349 @@ int main (int argc, char** argv)
...
+      //MES:
+      double unc_MESup_first;
+      double unc_MESdw_first;
+      double unc_MESup_second;
+      double unc_MESdw_second;
+
+      unc_MESup_first = theBigTree.daughters_MESshiftup->at (firstDaughterIndex); // first daughter, DM 0
+      unc_MESdw_first = theBigTree.daughters_MESshiftdw->at (firstDaughterIndex); // first daughter, DM 0
+      unc_MESup_second = theBigTree.daughters_MESshiftup ->at (secondDaughterIndex); // second daughter, DM 0
+      unc_MESdw_second = theBigTree.daughters_MESshiftdw ->at (secondDaughterIndex); // second daughter, DM 0
```
Comment (camendola) [inline test/skimNtuple2018.cpp]: You probably just hinerited a mistake that I had made in skimNtuple2017, but can you fix the comments? 
// first daughter, DM 0 -> // first daughter, up
// first daughter, DM 0 -> // first daughter, dw
etc.

File: test/skimNtuple2018.cpp [hunk 3]
```diff
@@ -1949,251 +2221,349 @@ int main (int argc, char** argv)
...
-        }
+          if(DEBUG)
+            {
+              cout << "L1 match for first daughter: " << theBigTree.daughters_highestEt_L1IsoTauMatched->at(firstDaughterIndex) << endl;
+              cout << "L1 match for second daughter: " << theBigTree.daughters_highestEt_L1IsoTauMatched->at(secondDaughterIndex) << endl;
+            }
 
-        if (DEBUG) std::cout << "----> Final Trigger passed? " << triggerAccept << std::endl;
-        if (!triggerAccept) continue;
+          // L1 match for 2017 data
```
Comment (camendola) [inline test/skimNtuple2018.cpp]: Please remove this check of L1 match for 2017 data (until line 2443)

File: test/skimNtuple2018.cpp [hunk 4]
```diff
@@ -1949,251 +2221,349 @@ int main (int argc, char** argv)
...
+                  }
+                if (!passL1IsoTau32) passTrg = false;
+              }
+          //}
 
-        theSmallTree.m_pass_triggerbit = pass_triggerbit;
-        ec.Increment ("Trigger", EvtW); // for data, EvtW is 1.0
-        if (isHHsignal && pairType == genHHDecMode) ecHHsig[genHHDecMode].Increment ("Trigger", EvtW);
-      }
+          // !! FIXME !! --> update the trigger bits to the right paths
```
Comment (camendola) [inline test/skimNtuple2018.cpp]: Please remove these mu prescale fixes (they are specific to 2017), until line 2457

**Comment by camendola** [comment]
@jonamotta also, can you please help me crosscheck that **only** the changes to skimNtuple2017.cpp displayed [here](https://github.com/francescobrivio/KLUBAnalysis/pull/9/files) are introduced?
Nothing 2017 specific should be propagated to the 2018 skimmer (for now I only spotted two in the trigger part that are 2017 specific); the DNN part of the code can stay.

File: test/skimNtuple2018.cpp [hunk 5]
```diff
@@ -1949,251 +2221,349 @@ int main (int argc, char** argv)
...
-          {
-            trigReader.listETau(triggerbit, matchFlag1, matchFlag2, trgNotOverlapFlag, goodTriggerType1, goodTriggerType2);
-          }
-          if(pairType == 2)//TauTau
-          {
-            trigReader.listTauTau(triggerbit, matchFlag1, matchFlag2, trgNotOverlapFlag, goodTriggerType1, goodTriggerType2);
-            trigReader.listVBF(triggerbit, matchFlag1, matchFlag2, trgNotOverlapFlag, goodTriggerType1, goodTriggerType2, tlv_firstLepton.Pt(), tlv_firstLepton.Eta(), tlv_secondLepton.Pt(), tlv_secondLepton.Eta());
-          }
-        }
+          if(DEBUG)
```
Comment (camendola) [inline test/skimNtuple2018.cpp:2426]: Please remove this debug parenthesis (until 2430)

File: test/skimNtuple2018.cpp [hunk 6]
```diff
@@ -5103,6 +5678,292 @@ int main (int argc, char** argv)
...
+      }
+
+    // Update tree and delete readers
+    treenew->Write ("", TObject::kOverwrite) ;
+    in_file->Close();
+  } // END NEW DNN
+*/
+
   cout << "... SKIM finished, exiting." << endl;
+  cout << "... CIAO." << endl;
```
Comment (camendola) [inline test/skimNtuple2018.cpp]: Remove your ciao :)

File: test/skimNtuple2018.cpp [hunk 7]
```diff
@@ -5103,6 +5678,292 @@ int main (int argc, char** argv)
 
   } // End new BDT
 
+/*
```
Comment (camendola) [inline test/skimNtuple2018.cpp]: I know I had told you to comment it out for the moment, but can you uncomment this DNN part?

File: test/skimNtuple2018.cpp [hunk 8]
```diff
@@ -47,8 +49,16 @@
 #include <boost/algorithm/string/replace.hpp>
 #include <boost/format.hpp>
 #include <Math/VectorUtil.h>
+#include <Math/LorentzVector.h>
+#include <Math/PxPyPzM4D.h>
+
+// DNN
```
Comment (camendola) [inline test/skimNtuple2018.cpp:55]: I know I had told you to comment it out for the moment, but can you uncomment this DNN part?
https://github.com/camendola/KLUBAnalysis/pull/34/files#r417865527

File: test/skimNtuple2018.cpp [hunk 9]
```diff
@@ -2998,1401 +3370,1592 @@ int main (int argc, char** argv)
...
+              theSmallTree.m_bjet2_mass_raw_jetdown.push_back(tlv_secondBjet_raw_jetdown[isource].M());
+              theSmallTree.m_bjet2_pt_raw_jetup.push_back(tlv_secondBjet_raw_jetup[isource].Pt());
+              theSmallTree.m_bjet2_pt_raw_jetdown.push_back(tlv_secondBjet_raw_jetdown[isource].Pt());
+              // variations propagated to bH
+              tlv_bH_raw_jetup.push_back(tlv_firstBjet_raw_jetup.at(isource) + tlv_secondBjet_raw_jetup.at(isource)) ;
+              tlv_bH_raw_jetdown.push_back(tlv_firstBjet_raw_jetdown.at(isource) + tlv_secondBjet_raw_jetdown.at(isource));
+              theSmallTree.m_bH_mass_raw_jetup.push_back((tlv_bH_raw_jetup[isource]).M());
+              theSmallTree.m_bH_mass_raw_jetdown.push_back((tlv_bH_raw_jetdown[isource]).M());
+              theSmallTree.m_bH_pt_raw_jetup.push_back((tlv_bH_raw_jetup[isource]).Pt());
+              theSmallTree.m_bH_pt_raw_jetdown.push_back((tlv_bH_raw_jetup[isource]).Pt());
```
Comment (camendola) [inline test/skimNtuple2018.cpp]: oops, this should be 
`theSmallTree.m_bH_pt_raw_jetdown.push_back((tlv_bH_raw_jetdpwn[isource]).Pt());`
(my mistake, I'll also fix it in skimNtuple2017).
Heads up to @francescobrivio (can't tag dzuolo), there's probably the same mistake in skimNtuple2016 if copied from my PR

Comment (francescobrivio) [reply test/skimNtuple2018.cpp]: nice catch @camendola ok we will fix it on the 2016 PR as well

File: test/skimNtuple2018.cpp [hunk 10]
```diff
@@ -2998,1401 +3370,1592 @@ int main (int argc, char** argv)
...
+          theSmallTree.m_bH_mass_raw_jetup2    = (tlv_bH_raw_jetdown[1]).M();
+          theSmallTree.m_bH_mass_raw_jetup3    = (tlv_bH_raw_jetdown[2]).M();
+          theSmallTree.m_bH_mass_raw_jetup4    = (tlv_bH_raw_jetdown[3]).M();
+          theSmallTree.m_bH_mass_raw_jetup5    = (tlv_bH_raw_jetdown[4]).M();
+          theSmallTree.m_bH_mass_raw_jetup6    = (tlv_bH_raw_jetdown[5]).M();
+          theSmallTree.m_bH_mass_raw_jetup7    = (tlv_bH_raw_jetdown[6]).M();
+          theSmallTree.m_bH_mass_raw_jetup8    = (tlv_bH_raw_jetdown[7]).M();
+          theSmallTree.m_bH_mass_raw_jetup9    = (tlv_bH_raw_jetdown[8]).M();
+          theSmallTree.m_bH_mass_raw_jetup10   = (tlv_bH_raw_jetdown[9]).M();
+          theSmallTree.m_bH_mass_raw_jetup11   = (tlv_bH_raw_jetdown[10]).M();
```
Comment (jonamotta) [inline test/skimNtuple2018.cpp]: Same kind of error with the mixing of up/down here @camendola & @francescobrivio , I don't know if it's the same in 2016 and 2017 but better to check

Comment (camendola) [reply test/skimNtuple2018.cpp]: Indeed, thanks, fixing in 2017 as well!

File: test/skimNtuple2018.cpp (hunk 5 above)
Comment (camendola) [reply test/skimNtuple2018.cpp:2426]: @jonamotta forgot this one

**Pull request merged**